### PR TITLE
better representation of i24

### LINKED ISSUE #2: Consistency, Endianness and Fallible Operations
In the current version of the crate, endianness is not handled very well and there isn't great consistency of behaviour across architectures. This mixed with unclear behaviour from the implementation of ``From``  there needs to be an overhaul of how i24s are created with care taken for endianess depending on the system. 

Some changes will include:
- ``TryFrom`` instead of ``Try`` 
- cfg compile flags

### PR DESCRIPTION
**Pull request by vrtgs**
better represent i24 in memory (should be really fast now)

fixes #2 #3

### CODE REVIEW

**Comment by vrtgs** [comment]
if only this was u24, then we would have had insanely fast division too😔

**Comment by jmg049** [comment]
Gonna have more of a read of the commit later today but it looks amazing. Thanks.

I have ask though, why?

**Comment by vrtgs** [comment]
> I have ask though, why?

I mean people expect statically sized integers to be fast, and I enjoyed doing bit magic on the type, and I have always wanted to do something exactly like this but felt there was no need for it, and well now I have found a reason for it

**Comment by jmg049** [comment]
Amazing. It's along the lines of why I implemented it in the first place. I had never done anything with close to native types but my interest has been growing and when the opportunity came up for implementing 24-bit integers for wavers I jumped at it.

Thanks again, I'll get around to looking at it more thoroughly soon.

**Comment by vrtgs** [comment]
alright with this I am sure this is complete, feel free to ask me about anything in the code, although I tried to add comments to clear up any part that isn't clear

**Comment by jmg049** [comment]
I'm reading through it when I have time. So far it's all good. It's all fairly self explanatory and the comments fill in the rest. 

I promise to get around as soon as I can. Currently a bank holiday where I am and also have plenty of my main work to be doing.

File: src/repr.rs [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    data: [u8; 3],
+    // most significant byte at the end
+    most_significant_byte: ZeroByte,
+}
+
+#[cfg(target_endian = "big")]
+type I24Repr = BigEndianI24Repr;
+
+#[cfg(target_endian = "little")]
+pub(super) type I24Repr = LittleEndianI24Repr;
```
Comment (jmg049) [inline src/repr.rs]: Is there a specific reason that everything is pub and super except for the type I24Repr when the system is big endian?

**Review by jmg049 — COMMENTED** [review]
Just one comment which is a question on the I24Repr. Looks really good to me, I think it's all self-explanatory to someone actually interested enough in looking through the code. Docs look good too. 

Really can't say thanks enough. Happy to approve after the question gets answered (it's super minor)

File: src/repr.rs (hunk 1 above)
Comment (vrtgs) [reply src/repr.rs]: no thats a mistake😅

**Comment by vrtgs** [comment]
alright latest commit should have fixed your concerns, and really it was no problem, I saw the post on reddit and saw thats cool, and I was happy to help, and work on something like this too :>

**Comment by vrtgs** [comment]
the tests were a bit of a mess, had to clean those up too

**Comment by jmg049** [comment]
That's that. The code is much much better shape and I will make it no_std and have another look at things before publishing to crates. Thanks again

**Pull request merged**